### PR TITLE
[FEATURE] Add support to update record fields

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds backend support to update record fields.

**Type of change**

- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: .github/workflows/argilla.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
   build:
     services:
       argilla-server:
-        image: argilladev/argilla-hf-spaces:develop
+        image: argilladev/argilla-hf-spaces:pr-5685
```
Comment (frascuchon) [inline .github/workflows/argilla.yml]: TODO: Change before merge
```suggestion
        image: argilladev/argilla-hf-spaces:develop
```

File: argilla-server/src/argilla_server/api/schemas/v1/records.py [hunk 2]
```diff
@@ -205,15 +200,20 @@ def prevent_nan_values(cls, metadata: Optional[Dict[str, Any]]) -> Optional[Dict
 
         return {k: v for k, v in metadata.items() if v == v}  # By definition, NaN != NaN
 
+    def is_set(self, attribute: str) -> bool:
+        return attribute in self.__fields_set__
 
-class RecordUpdateWithId(RecordUpdate):
-    id: UUID
+    def has_changes(self) -> bool:
+        return self.dict(exclude_unset=True) != {}
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/api/schemas/v1/records.py]: Should we use `model_dump` instead?

```suggestion
        return self.model_dump(exclude_unset=True) != {}
```

File: argilla-server/src/argilla_server/contexts/records.py [hunk 3]
```diff
@@ -113,3 +124,111 @@ def _record_by_dataset_id_query(
...
+    deleted_record_events_v1 = []
+    for record in records:
+        deleted_record_events_v1.append(
+            await build_record_event_v1(db, RecordEvent.deleted, record),
+        )
+
+    records = await Record.delete_many(
+        db,
+        conditions=params,
+        autocommit=True,
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/contexts/records.py:228]:
```suggestion
```

**Review by jfcalvo — APPROVED** [review]

File: argilla-server/src/argilla_server/contexts/records.py (hunk 3 above)
Comment (frascuchon) [reply argilla-server/src/argilla_server/contexts/records.py:228]: I prefer to explicitly set the commit flag to True, even if it is the default value. It's a bit weird and probably dangerous to set commit to True by default.

**Comment by davidberenstein1957** [comment]
@nataliaElv @frascuchon do you remember if there was a dedicated issues for this? I feel there should be but I could not find and I wanted to ping users there to mention we added this support.

**Comment by davidberenstein1957** [comment]
I believe @cceyda once requested this too :)

**Pull request merged**